app/actions/configurations.py

```diff
@@ -0,0 +1,58 @@
+import pydantic
+
+from app.actions.client import Station
+from app.actions.core import AuthActionConfiguration, PullActionConfiguration, InternalActionConfiguration, ExecutableActionMixin
+from app.services.errors import ConfigurationNotFound
+from app.services.utils import find_config_for_action, UIOptions, FieldWithUIOptions
+
+
+class AuthenticateConfig(AuthActionConfiguration, ExecutableActionMixin):
+    key: pydantic.SecretStr = pydantic.Field(..., format="password")
+
+
+class PullObservationsConfig(PullActionConfiguration):
+    default_lookback_days: int = FieldWithUIOptions(
+        15,
+        title="Default Lookback Days",
+        description="Initial number of days to look back for observations Min: 1",
+        ge=1,
+        le=30,
+        ui_options=UIOptions(
+            widget="range",  # This will be rendered ad a range slider
+        )
+    )
+
+
+class PullStationHistoryConfig(PullActionConfiguration, InternalActionConfiguration):
```

Comment:
Remove `PullActionConfiguration` since the related sub-action will be triggered by the main action?
```suggestion
class PullStationHistoryConfig(InternalActionConfiguration):
```